### DISCUSSION TITLE
[Question] Determine artifact available versions?

### DISCUSSION THREAD

**Discussion by BoykoAlex** [eclipse-m2e/m2e-core#1213] [Q&A] [answered]
Is there a way with m2e core/maven runtime to obtain all versions of an artifact (either `org.apache.maven.artifact.Artifact` or group and artifact id instead) given a Maven project context (i.e. artifact repositories specified for the maven project)

I thought I'd go with org.apache.maven.project.MavenProject, get its list of `ArtifactRepository` and then call on each repo `List<String> findVersions(Artifact artifact)`. This always return empty list :-\

I found `org.eclipse.m2e.core.ui.internal.search.util.SearchEngine` interface but could find any concrete implementation of it. 

Anything you could recommend or give some pointers?

Thanks in advance!

**Comment by laeubi** [comment, marked as answer]
There is a demo using the resolver here:

https://github.com/apache/maven-resolver/blob/master/maven-resolver-demos/maven-resolver-demo-snippets/src/main/java/org/apache/maven/resolver/examples/FindAvailableVersions.java

You can find an example in m2eclipse here:
https://github.com/eclipse-m2e/m2e-core/blob/19a4aba3262bacb28ded789a81863e2780229cad/org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/MavenTargetLocation.java#L417-L444

**Comment by BoykoAlex** [comment]
Thanks very much! Precisely what i was looking for!
